syft/source/directorysource/directory_source.go

```diff
@@ -145,12 +146,16 @@ func (s *directorySource) FileResolver(_ source.Scope) (file.Resolver, error) {
 		// this should be the only file resolver that might have overlap with where files are cached
 		exclusionFunctions = append(exclusionFunctions, excludeCachePathVisitors()...)
 
-		res, err := fileresolver.NewFromDirectory(s.config.Path, s.config.Base, exclusionFunctions...)
-		if err != nil {
-			return nil, fmt.Errorf("unable to create directory resolver: %w", err)
+		if s.config.Unindexed {
+			res := fileresolver.NewFromRootedUnindexedDirectory(s.config.Path, s.config.Base)
+			s.resolver = res
```

Comment:
nit: why not just `s.resolver = fileresolver.NewFromRootedUnindexedDirectory(s.config.Path, s.config.Base)`